spec/integration/apply_spec.rb

```diff
@@ -4,266 +4,288 @@
+        expect(result).to include('kind' => 'bolt/apply-failure')
+        error = result['details']['result_set'][0]['value']['_error']
+        expect(error['kind']).to eq('bolt/resource-failure')
+        expect(error['msg']).to match(/Resources failed to apply/)
+      end
+
+      it 'applies a notify and ignores local settings' do
+        command = 'echo environment=doesnotexist > /etc/puppetlabs/puppet/puppet.conf'
+        run_cli_json(%W[command run #{command} -t agent_targets], project: project)
+
+        result = run_cli_json(%w[plan run basic::class -t agent_targets], project: project)
+        expect(result).not_to include('kind')
         expect(result[0]).to include('status' => 'success')
-        expect(result[0]['value']['version']).to match(/^6/)
+        expect(result[0]['value']['_output']).to eq('changed: 1, failed: 0, unchanged: 0 skipped: 0, noop: 0')
+        resources = result[0]['value']['report']['resource_statuses']
+        expect(resources).to include('Notify[hello world]')
+      end
+
+      it 'respects _run_as on a plan invocation' do
+        user = conn_info('ssh')[:user]
+        logs = run_cli_json(%W[plan run basic::run_as_apply user=#{user} -t agent_targets], project: project)
+        expect(logs.first['message']).to eq(conn_info('ssh')[:user])
       end
 
-      it 'runs a ruby task' do
-        with_tempfile_containing('inventory', YAML.dump(agent_version_inventory), '.yaml') do |inv|
-          results = run_cli_json(%W[task run basic::ruby_task --targets agent_targets
-                                    --modulepath #{modulepath} --inventoryfile #{inv.path}])
-          results['items'].each do |result|
-            expect(result['status']).to eq('success')
-            expect(result['value']).to eq('ruby' => 'Hi')
-          end
+      it 'respects show_diff configuration' do
+        diff_string = <<~DIFF
+          @@ -1 +1 @@
+          -Silly string
+          \\ No newline at end of file
+          +Silly string (get it?)
+          \\ No newline at end of file
+        DIFF
+
+        results = run_cli_json(%w[plan run settings::show_diff -t agent_targets], project: project)
+
+        results.each do |result|
+          expect(result['status']).to eq('success')
+          expect(result['value']['report']['logs'][0]['message']).to include(diff_string)
         end
       end
 
-      it 'runs an apply plan' do
-        with_tempfile_containing('inventory', YAML.dump(agent_version_inventory), '.yaml') do |inv|
-          results = run_cli_json(%W[plan run basic::notify --targets agent_targets
-                                    --modulepath #{modulepath} --inventoryfile #{inv.path}])
-          results.each do |result|
-            expect(result['status']).to eq('success')
-            report = result['value']['report']
-            expect(report['resource_statuses']).to include("Notify[Apply: Hi!]")
-          end
+      it "applies a manifest" do
```

Comment:
I think it's still useful to wrap this in a `bolt apply command` context so we can better pinpoint failures.